src/webgpu/web_platform/copyToTexture/canvas.spec.ts

```diff
@@ -214,17 +317,131 @@ g.test('copy_contents_from_canvas')
 
     // Construct expected value for different dst color format
     const dstBytesPerPixel = kTextureFormatInfo[dstColorFormat].bytesPerBlock;
+    const format: RegularTextureFormat =
+      dstColorFormat === 'rgba8unorm-srgb' || dstColorFormat === 'bgra8unorm-srgb'
+        ? dstColorFormat === 'rgba8unorm-srgb'
+          ? 'rgba8unorm'
+          : 'bgra8unorm'
+        : dstColorFormat;
+    dstColorFormat;
+
+    // For 2d canvas, get expected pixels with getImageData(), which returns origin
+    // values(not applied premultipy alpha) always.
```

Comment:
nit: "origin" (noun) should be "original" (adjective).

Although actually I wouldn't call them "original" values. Canvas normally operates in premultiplied space. So we can just say:
```suggestion
    // For 2d canvas, get expected pixels with getImageData(), which returns unpremultiplied values.
```